sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/NonAzureOpenAIAsyncClientTest.java

```diff
@@ -47,17 +49,16 @@ public void getCompletions(HttpClient httpClient, OpenAIServiceVersion serviceVe
     @Disabled("onError(com.fasterxml.jackson.databind.exc.MismatchedInputException: Missing required creator property 'usage'")
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.ai.openai.TestUtils#getTestParameters")
-    public void testGetCompletionsStream(HttpClient httpClient, OpenAIServiceVersion serviceVersion) {
+    public void getCompletionsStream(HttpClient httpClient, OpenAIServiceVersion serviceVersion) {
         client = getNonAzureOpenAIAsyncClient(httpClient);
-        getCompletionsRunner((modelId, prompt) -> {
-            StepVerifier.create(client.getCompletionsStream(modelId, new CompletionsOptions(prompt)).last())
-                .assertNext(completions -> {
-                    assertNotNull(completions.getId());
-                    assertNotNull(completions.getChoices());
-                    assertFalse(completions.getChoices().isEmpty());
-                    assertNotNull(completions.getChoices().get(0).getText());
-                })
-                .verifyComplete();
+        getCompletionsRunner((deploymentId, prompt) -> {
+            StepVerifier.create(client.getCompletionsStream(deploymentId, new CompletionsOptions(prompt)))
+                .recordWith(ArrayList::new)
+                .thenConsumeWhile(chatCompletions -> true)
+                .consumeRecordedWith(messageList -> {
+                    assertTrue(messageList.size() > 1);
+                    messageList.forEach(OpenAIClientTestBase::assertCompletionsStream);
+                }).verifyComplete();
```

Comment:
We can move the assertions to `thenConsumeWhile()`.

```suggestion
                .recordWith(ArrayList::new)
                .thenConsumeWhile(OpenAIClientTestBase::assertCompletionsStream)
                .consumeRecordedWith(messageList -> assertTrue(messageList.size() > 1))
                .verifyComplete();
```

Reply:
But I see your point, if we assert false, we want the flux to be interrupted. I will move the assertion to that lambda like you suggest, but I don't see a way around returning a nasty hardcoded `true` somewhere 🤔